src/main/scala/mesosphere/marathon/core/storage/migration/MigrationTo1_3_PersistentStore.scala

```diff
@@ -0,0 +1,114 @@
+package mesosphere.marathon.core.storage.migration
+
+// scalastyle:off
+import akka.Done
+import akka.stream.Materializer
+import akka.stream.scaladsl.Source
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.marathon.core.storage.LegacyStorageConfig
+import mesosphere.marathon.core.storage.repository.{ AppRepository, DeploymentRepository, GroupRepository, Repository, TaskFailureRepository, TaskRepository, VersionedRepository }
+import mesosphere.marathon.metrics.Metrics
+import mesosphere.marathon.state.{ AppDefinition, Group, MarathonTaskState, TaskFailure }
+import mesosphere.marathon.upgrade.DeploymentPlan
+
+import scala.async.Async.{ async, await }
+import scala.concurrent.{ ExecutionContext, Future }
+// scalastyle:on
+
+/**
+  * Migration from Legacy Storage in 1.2 to the new Persistence Storage in 1.3
+  *
+  * Does nothing unless Legacy Storage and New Storage are configured.
+  */
+class MigrationTo1_3_PersistentStore(migration: Migration)(implicit
+  executionContext: ExecutionContext,
+    mat: Materializer,
+    metrics: Metrics) extends StrictLogging {
+  def migrate(): Future[Done] = async {
+    val legacyStore = await(migration.legacyStoreFuture)
+    // This migration only goes from legacy storage -> new storage.
```

Comment:
I was trying to point out that this one only makes sense if we go from `legacy->new`, it doesn't make sense in `new->new'`. I made those be strongly differentiated.